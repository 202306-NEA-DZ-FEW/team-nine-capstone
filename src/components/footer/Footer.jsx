import { getAuth } from "firebase/auth";
import Link from "next/link";
import { useTranslation } from "next-i18next";
import React from "react";

function Footer() {
    const auth = getAuth();
    const user = auth.currentUser;

    const { t } = useTranslation("common");
    return (
        <footer className='bottom-0 w-screen mx-auto  text-center bg-gray-800 text-white backdrop:blur-2xl'>
            <div className='px-6  md:py-14 xl:pt-10 xl:pb-10 py-6'>
                <h2 className='font-semibold text-3xl xl:text-3xl leading-snug'>
                    {t("footer.hook")}
                </h2>
                <div className='mt-10 xl:mt-10'>
                    <nav className='flex flex-wrap justify-center text-lg font-medium'>
                        <div className='px-5 py-2'>
                            <Link href='/about'>{t("About")}</Link>
                        </div>
                        <div className='px-5 py-2'>
                            <Link href='/termsAndConditions'>
                                {t("Terms of Use")}
                            </Link>
                        </div>
                        <div className='px-5 py-2'>
                            <Link href='/events'>{t("Events")}</Link>
                        </div>
                        <div className='px-5 py-2'>
                            <Link href='/authentication/signIn'>
                                {user ? t("Account") : t("Sign In")}
                            </Link>
                        </div>
                        <div className='px-5 py-2'>
                            <Link href='/authentication/signUp'>
                                {user ? t("Your Events") : t("Sign Up")}
                            </Link>
                        </div>
                    </nav>
                    <hr className='text-black' />
                    <p className='mt-7 font-medium'>
                        Â© Re:Coded DZ NEA FEW 2023
                    </p>
                </div>
            </div>
        </footer>
    );
}

export default Footer;
