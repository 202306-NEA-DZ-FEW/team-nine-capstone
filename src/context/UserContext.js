import React, { createContext, useContext, useEffect, useState } from "react";

import { auth } from "@/lib/firebase/controller";

const UserContext = createContext();

export function UserProvider({ children }) {
    const [user, setUser] = useState(null);

    useEffect(() => {
        const isLogged = auth.onAuthStateChanged((authUser) => {
            if (authUser) {
                // User is signed in
                setUser(authUser);
            } else {
                // User is signed out
                setUser(null);
            }
        });

        // Clean up the subscription
        return () => isLogged();
    }, []);

    return (
        <UserContext.Provider value={{ user, setUser }}>
            {children}
        </UserContext.Provider>
    );
}

export function useUser() {
    return useContext(UserContext);
}
