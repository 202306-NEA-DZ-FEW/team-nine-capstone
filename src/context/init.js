import React from "react";

function init() {
    return <div>user context and other stuff</div>;
}

export default init;
