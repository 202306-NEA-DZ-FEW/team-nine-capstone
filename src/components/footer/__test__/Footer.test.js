import renderer from "react-test-renderer";

import { useUser } from "@/context/UserContext";

import Footer from "../Footer";

// Mock Firebase
jest.mock("@firebase/app", () => ({
    __esModule: true,
    initializeApp: jest.fn(),
    getApps: jest.fn(() => [{}]),
    app: jest.fn(),
}));

jest.mock("@/lib/firebase/controller.js", () => ({
    __esModule: true,
    getAuth: jest.fn(),
    onAuthStateChanged: jest.fn(),
    signInWithEmailAndPassword: jest.fn(),
    signOut: jest.fn(),
}));

jest.mock("src/context/UserContext.js");

describe("Footer", () => {
    it("renders correctly", () => {
        // Mock the useUser hook to provide a dummy implementation
        useUser.mockReturnValue({ user: { uid: "dummyUserId" } });

        const tree = renderer.create(<Footer />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
