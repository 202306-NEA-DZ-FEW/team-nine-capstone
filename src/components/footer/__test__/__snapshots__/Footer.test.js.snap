// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="bottom-0 w-screen mx-auto  text-center bg-gray-800 text-white backdrop:blur-2xl"
>
  <div
    className="px-6  md:py-14 xl:pt-10 xl:pb-10 py-6"
  >
    <h2
      className="font-semibold text-3xl xl:text-3xl leading-snug"
    >
      footer.hook
    </h2>
    <div
      className="mt-10 xl:mt-10"
    >
      <nav
        className="flex flex-wrap justify-center text-lg font-medium"
      >
        <div
          className="px-5 py-2"
        >
          <a
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            About
          </a>
        </div>
        <div
          className="px-5 py-2"
        >
          <a
            href="/termsAndConditions"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Terms of Use
          </a>
        </div>
        <div
          className="px-5 py-2"
        >
          <a
            href="/events"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Events
          </a>
        </div>
        <div
          className="px-5 py-2"
        >
          <a
            href="/authentication/signIn"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Account
          </a>
        </div>
        <div
          className="px-5 py-2"
        >
          <a
            href="/authentication/signUp"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Your Events
          </a>
        </div>
      </nav>
      <hr
        className="text-black"
      />
      <p
        className="mt-7 font-medium"
      >
        Â© Re:Coded DZ NEA FEW 2023
      </p>
    </div>
  </div>
</footer>
`;
